client/state/data-layer/utils.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * External dependencies
+ */
+import {
+	isArray,
+	mergeWith,
+} from 'lodash';
+
+const concatHandlers = ( left, right ) =>
+	isArray( left )
+		? left.concat( right )
+		: undefined;
+
+export const mergeHandlers = ( ...handlers ) =>
+	handlers.length > 1
+		? mergeWith( {}, ...handlers, concatHandlers )
+		: handlers.shift();
```

Comment:
> Would handlers[ 0 ] (head( handlers )) be clearer?

not to me at all. `Array.prototype.shift()` is builtin to JavaScript and does one very straightforward thing. I think changing it to the proposed is not only adding a strange formulation but also wrong. the proposed change could have just been `handlers[ 0 ]` and that's functionally identical to `handlers.shift()`; I just prefer `.shift()` in many cases is all.